Yoast/YoastSEO.js
# Refactor precedence exceptions

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the Chrome browser tab would crash on Windows when a French or Italian text contains sentences in passive voice. Thanks for the input @CarloCannas.

## Relevant technical choices:

* Refactor of `directPrecedenceExceptionWithoutRegex`.
  - Changed it to handle sentence parts and participle as a word to make it easier to implement in our current code.
  - Added support to the other direct precedence languages too.
  - Changed the `Participle` classes of those languages (Dutch, English, French, Italian and Spanish -- Polish already used it) to use this version instead of the regex one.
* Added `precedenceExceptionWithoutRegex`.
  - Changed the `Participle` classes that used `precedenceExpection` to use this version instead (English, French, Italian and Spanish).
* Added a string check to the `setParticiple` function of the `Participle` base class, that throws an error when trying to set any other type. This to make it even more clear non-string is not supported, as I saw tests using null as participle and still calling the `directPrecedenceException` (withoutRegex) function. Which will error. Removed these offending lines in the specs (Italian and Spanish empty tests).
* Added French function word to the list `l'avoir`. This is due to the word boundary regexes not being applied to the function words anymore. Which is very much intentional and the point of this fix. However, a side effect is there might be more cases like this that did get matched before, that no longer work.

## Test instructions

This PR can be tested by following these steps:

On Windows 10, in the Chrome browser.
* Change WP Site Language to French (Français).
* Create a new Post/Page and enter French content: `Il a été le passivé. Il a été a passivé.`. This should no longer crash the browser tab. Possibly publish and change the text. Try a bit harder to make it fail. In my tests to see the issue it was not 100% always crashing the tab.
* Change WP Site Language to Italian (Italiano).
* Create a new Post/Page and enter Italian content: `Questa pagina è stata la modificata. Questa pagina è stata sto modificata.`. This should no longer crash the browser tab. Possibly publish and change the text. Try a bit harder to make it fail. In my tests to see the issue it was not 100% always crashing the tab.

* Repeat steps in Opera browser.

Other passive voices where also touched. Please take a look if they still behave as before.

Passive voice: Direct precedence exceptions:
* Dutch
* English
* French
* ~~German~~ -- already done differently (not with a regex).
* Italian
* ~~Polish~~ -- already changed in https://github.com/Yoast/YoastSEO.js/pull/1812
* Spanish

Passive voice: Precedence exceptions:
* English
* French
* Italian
* Spanish

Hotfix version.
Related to https://github.com/Yoast/YoastSEO.js/pull/1910
